docs/user-guide/kubeconfig-file.md

```diff
@@ -94,7 +94,89 @@ users:
     client-key: path/to/my/client/key
 ```
 
-### Building your own kubeconfig file
+### Breakdown/explanation of components
+
+#### cluster
+
+```
+clusters:
+- cluster:
+    api-version: v1
+    server: http://cow.org:8080
+  name: cow-cluster
+```
+
+A `cluster` contains endpoint data for a kubernetes cluster. This includes the fully
+qualified url for the kubernetes apiserver, as well as the cluster's certificate
+authority or `insecure-skip-tls-verify: true`, if the cluster does not support tls.
+A `cluster` has a name (nickname) which acts as a dictionary key for the cluster
+within this kubeconfig file. You can add or modify `cluster` entries using
+[`kubectl config set-cluster`](kubectl/kubectl_config_set-cluster.md).
+
+#### user
+
+```
+users:
+- name: blue-user
+  user:
+    token: blue-token
+- name: green-user
+  user:
+    client-certificate: path/to/my/client/cert
+    client-key: path/to/my/client/key
+```
+
+A `user` defines client credentials for authenticating to a kubernetes cluster. A
+`user` has a name (nickname) which acts as its key within the list of user entries
+after kubeconfig is loaded/merged. You can add or modify `user` entries using
+[`kubectl config set-credentials`](kubectl/kubectl_config_set-credentials.md).
```

Comment:
Also,  username/password and token are mutually exclusive, but client-certs and keys can be combined with them.